docs/interpolation/Profile-Interpolator/ProfileInterpolator.rst

```diff
@@ -0,0 +1,106 @@
+
+.. _ProfileInterpolator:
+
+###########################
+Profile Interpolation
+###########################
+
+
+We demonstrate the profile interpolator which delivers predictions 
+on internal stellar structure at the end of MESA evolution. 
+To use the profile interpolator we firstimport the ``ProfileInterpolator`` 
+object from the POSYDON library.
+
+.. code-block:: python
+
+  from posydon.interpolation.profile_interpolation import ProfileInterpolator, CompileData
+
+
+Extracting Profile Data from .h5 Files for Training Interpolators
+===============================
+
+To extract profile data from existing grids, we pass arguments 
+``train_path`` and ``valid_path`` pointing to .h5 files 
+containing instances of the ``PSyGrid`` class from which the
+training and testing grids can be loaded. These can be constructed 
+by running ones own simulations or by loading a precomputed simulation 
+stored in the data directory of the POSYDON repository.
+The ``profile_names`` argument denotes which profiles will be extracted 
+from the grid and saved to ``filename`` using the function ``save``. 
+
+.. code-block:: python
+
+  compiler = CompileData(train_data = '/path/to/training/grid.h5',
+                         valid_data = '/path/to/testing/grid.h5',
+                         profile_names=['radius','logRho',
+                         'x_mass_fraction_H','y_mass_fraction_He',
+                         'z_mass_fraction_metals','omega','energy'])
+  compiler.save(filename = '/path/for/profiles/datafile.pkl')
+  
+  
+Loading a Pretrained Interpolator
+===============================
+
+To load a pretrained interpolator we need to pass the optional
+``load_interpolator`` argument to the ``ProfileInterpolator`` 
+instance which specifies the path to a .pkl file where the 
+pretrained interpolator can be loaded from. 
+
+.. code-block:: python
+
+  model = ProfileInterpolator(load_interpolator = "/path/to/profile/interpolator.pkl")
```

Comment:
we should be consistent with how we load pertained interpolators across all of our interpolators. In the IFInterpolator we have a method called load that we call after constructing the class and the argument that points to where the interpolator is saved is called filename.